packages/twenty-server/src/engine/core-modules/twenty-config/cache/config-cache.service.ts

```diff
@@ -0,0 +1,130 @@
+import { Injectable, OnModuleDestroy } from '@nestjs/common';
+
+import { CONFIG_VARIABLES_CACHE_SCAVENGE_INTERVAL } from 'src/engine/core-modules/twenty-config/constants/config-variables-cache-scavenge-interval';
+import { CONFIG_VARIABLES_CACHE_TTL } from 'src/engine/core-modules/twenty-config/constants/config-variables-cache-ttl';
+
+import {
+  ConfigCacheEntry,
+  ConfigKey,
+  ConfigNegativeCacheEntry,
+  ConfigValue,
+} from './interfaces/config-cache-entry.interface';
+
+@Injectable()
+export class ConfigCacheService implements OnModuleDestroy {
+  private readonly valueCache: Map<ConfigKey, ConfigCacheEntry<ConfigKey>>;
+  private readonly negativeLookupCache: Map<
+    ConfigKey,
+    ConfigNegativeCacheEntry
+  >;
+  private cacheScavengeInterval: NodeJS.Timeout;
+
+  constructor() {
+    this.valueCache = new Map();
+    this.negativeLookupCache = new Map();
+    this.startCacheScavenging();
+  }
+
+  get<T extends ConfigKey>(key: T): ConfigValue<T> | undefined {
+    const entry = this.valueCache.get(key);
+
+    if (entry && !this.isCacheExpired(entry)) {
+      return entry.value as ConfigValue<T>;
+    }
+
+    return undefined;
+  }
+
+  getNegativeLookup(key: ConfigKey): boolean {
```

Comment:
The whole "GetNegative/SetNegative" is confusing to me at first read before digging into it. Is there a better naming you could use? To me NegativeLookup means the lookup will return no result which is kind of strange because you can't predict when calling the function?